java/code/src/com/redhat/rhn/manager/satellite/UpgradeCommand.java

```diff
@@ -168,6 +175,30 @@ public void upgrade() {
         }
     }
 
+    /**
+     * Encrypt the password of credentials with AES and a master password.
+     * This master password is autogenerated if it does not exist.
+     */
+    private void encryptCredentials() {
+        log.warn("Execute encryptCredentials");
+        try {
+            File mpw = new File(Config.getDefaultMasterPasswordFile());
```

Comment:
If we are migrating from an existing system how the master file is created? Instead of failing in here should we re-generate a new master file?

Reply:
oh, in the meantime I forgot I had this comment and I think I found where the file is created. It's on the file `[spacewalk/admin/spacewalk-startup-helper](https://github.com/uyuni-project/uyuni/pull/7840/files#diff-209785fca6714cd33a0feaaa410c709e23af5aaa36c314e71770a2698c38ce71) ?

Reply:
via systemd service during startup. The problem is, that tomcat cannot write in that directory. We can only read.